tests/storage/test_events.py

```diff
@@ -329,3 +329,63 @@ def test_do_not_prune_gap_if_not_dummy(self):
 
         # Check the new extremity is just the new remote event.
         self.assert_extremities([local_message_event_id, remote_event_2.event_id])
+
+
+class InvalideUsersInRoomCacheTestCase(HomeserverTestCase):
+    servlets = [
+        admin.register_servlets,
+        room.register_servlets,
+        login.register_servlets,
+    ]
+
+    def prepare(self, reactor, clock, homeserver):
+        self.state = self.hs.get_state_handler()
+        self.persistence = self.hs.get_storage().persistence
+        self.store = self.hs.get_datastore()
+
+    def test_remote_user_cache_invalidated(self):
+        """Test that if the server leaves a room the `get_users_in_room` cache
+        is invalidated for remote users.
+        """
```

Comment:
I think this is meant to be:

```suggestion
        """Test that if the server leaves a room the `get_rooms_for_user` cache
        is invalidated for remote users.
        """
```

Although I wonder if we should check the `get_users_for_room` one too while we're here?

Reply:
> Although I wonder if we should check the get_users_for_room one too while we're here?

I've written a test for it, and looks like it was being correctly invalidated.